keras/optimizers/optimizer_experimental/optimizer_test.py

```diff
@@ -281,6 +281,8 @@ def testSaveAndLoadOptimizerWithModel(self, optimizer_fn):
         [keras.layers.Input(shape=(1,)),
          keras.layers.Dense(1)])
     optimizer = optimizer_fn()
+    if type(optimizer) is adamw_new.AdamW:
```

Comment:
- Use `isinstance` instead of `type(...) is`
- This test checks that the code runs but does not check that the code is correct. Please add a correctness test.